core/shared/src/main/scala/zio/FiberId.scala

```diff
@@ -63,10 +63,10 @@ object FiberId {
   def apply(id: Int, startTimeSeconds: Int, location: Trace): FiberId =
     Runtime(id, startTimeSeconds * 1000L, location)
 
-  private[zio] def make(location: Trace)(implicit unsafe: Unsafe): FiberId.Runtime =
-    FiberId.Runtime(_fiberCounter.getAndIncrement(), java.lang.System.currentTimeMillis(), location)
-
-  private[zio] val _fiberCounter = new java.util.concurrent.atomic.AtomicInteger(0)
+  private[zio] def make(location: Trace)(implicit unsafe: Unsafe): FiberId.Runtime = {
+    val id = ThreadLocalRandom.current().nextInt(Int.MaxValue)
+    FiberId.Runtime(id, java.lang.System.currentTimeMillis(), location)
```

Comment:
not sure how much we care about order preserving, but isn't it possible to use N atomic counters (where N is number of cores) and then pick one by hashing the current thread id, and use the 'group id' as part of the fiber id, then the order is (roughly speaking) determined by the tuple(time, group, id within group)

Reply:
In theory we could make it work, but there are a couple of things we'd need to cater for which will add some penalty to performance. The biggest issue is that this approach would only work for `ZScheduler.Worker` threads. For any threads managed by another thread pool (virtual, blocking, etc) we'd have to fallback to using a globally initialized `AtomicInteger`.

I think that as long as we don't see a problem with losing strict fiber ordering for fibers spawned within the same millisecond (currently ordering only needed for zio-test), using a randomly generated ID might be our best option w.r.t performance.

Reply:
Let's keep this in mind in case we think it's necessary to preserve strict ordering of the fibers that are spawned within the same millisecond 👍

Reply:
Strict ordering is not required, IMO. It's a nice property to have, but performance is more important. 

The primary ordering by millisecond is sufficient.